src/flb_task.c

```diff
@@ -483,3 +483,33 @@ void flb_task_destroy(struct flb_task *task, int del)
     }
     flb_free(task);
 }
+
+struct flb_task_queue* flb_task_queue_create() {
+    struct flb_task_queue *tq;
+    tq = flb_malloc(sizeof(struct flb_task_queue));
+    if (!tq) {
+        flb_errno();
+        return NULL;
+    }
+    mk_list_init(&tq->pending);
+    mk_list_init(&tq->in_progress);
+    return tq;
+}
+
+void flb_task_queue_destroy(struct flb_task_queue *queue) {
+    struct flb_task_enqueued *queued_task;
+    struct mk_list *tmp;
+    struct mk_list *head;
+
+    mk_list_foreach_safe(head, tmp, &queue->pending) {
+        queued_task = mk_list_entry(head, struct flb_task_enqueued, _head);
+        flb_free(queued_task);
+    }
+
+    mk_list_foreach_safe(head, tmp, &queue->in_progress) {
+        queued_task = mk_list_entry(head, struct flb_task_enqueued, _head);
+        flb_free(queued_task);
```

Comment:
Since queued_task was allocated with flb_malloc, I think it is best for us to use flb_free to clean it up. Really the mk_list macros are just manipulating the _head pointer in the queued_task struct, and there's actually no special wrapper around the queued_task or added Monkey data inside it. What do you think?